src/time_listener.h

```diff
@@ -33,6 +33,12 @@ class TimeListener: virtual public Ider {
   ///
   /// @param time is the current simulation timestep
   virtual void Tock() = 0;
+
+  /// Simulation agents do their end-of-timestep decisions in the
```

Comment:
Perhaps some more info on what constitutes a 'decision' might be appropriate for this docstring.